src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -73,15 +93,25 @@ public override bool IsTriggerCharacter(SourceText text, int characterPosition,
 
         protected override ValueTuple<string, string> GetDisplayAndInsertionText(ISymbol symbol, AbstractSyntaxContext context)
         {
-            var insertionText = ItemRules.GetInsertionText(symbol, context);
+            var insertionText = GetInsertionText(symbol, context);
             var displayText = symbol.GetArity() == 0 ? insertionText : string.Format("{0}<>", insertionText);
 
             return ValueTuple.Create(displayText, insertionText);
         }
 
-        protected override CompletionItemRules GetCompletionItemRules()
+        private static CompletionItemRules s_importDirectiveRules =
```

Comment:
So we've lost lots of comments we used to have before.  Things like:

```
-                    // If the user is writing "using S" then the only commit characters are <dot> and
 -                    // <semicolon>, as they might be typing a using alias.
 -                    return ch == '.' || ch == ';';
```

We should preserve those.  This applies to *all* the completion item rules that we have in your change.